csrf_test.go

```diff
@@ -41,6 +41,27 @@ func TestProtect(t *testing.T) {
 	}
 }
 
+// TestExclude is a test to make sure that certain paths can be excluded from
+// the CSRF check.
+func TestExclude(t *testing.T) {
+	s := http.NewServeMux()
+	s.HandleFunc("/", testHandler)
+
+	p := Protect(testKey, Exclude("/"))(s)
```

Comment:
Please provide a robust set of table-driven tests - e.g. something like:

```go
var excludeTests = []struct{
    path string
    shouldExclude bool
}{
    {"/api/one", true},
    {"/admin", false},
}
```

The goal is to encompass a wide range of possible cases to prevent future regressions - e.g. if there is demand for more complex matching.